packages/alfa-act/src/rule.ts

```diff
@@ -24,22 +24,23 @@ const { flatMap, flatten, reduce } = Iterable;
 /**
  * @public
  */
-export abstract class Rule<I = unknown, T = unknown, Q = never>
+export abstract class Rule<I = unknown, T = unknown, Q = never, S = never>
```

Comment:
It feels like `S` should default to `T` since this is the main case we have. Would simplify things quite a bit in removing the need to specify it all over the place.